### PR TITLE
Remove unused AppTP Dax onboarding

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/1202279501986195/1202494492974085/f

### Description
We've stopped the AppTP retention study. We can now remove the AppTP Dax onboarding

### Steps to test this PR

_Test onboarding_
- [x] fresh install from this branch
- [x] verify the onboarding is the regular one and works as expected
- [x] enable AppTP and perform smoke test
- [x] smoke test in the browser

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2016** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2016" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2016?utm_source=stack-comment).

**Comment by aitorvs** [comment]
@anikiki I know this is not passing the JVM CI, it is because I need to do some param cleanup in the test. But the PR is ready to review

File: app/src/main/java/com/duckduckgo/app/onboarding/ui/page/vpn/VpnPermissionPage.kt [hunk 1]
```diff
@@ -1,215 +0,0 @@
...
-            .flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
-            .onEach { processCommand(it) }
-            .launchIn(lifecycleScope)
-    }
-
-    private fun processCommand(command: VpnPagesViewModel.Command) {
-        when (command) {
-            is ShowVpnConflictDialog -> launchVPNConflictDialog(false)
-            is ShowVpnAlwaysOnConflictDialog -> launchVPNConflictDialog(true)
-            is LeaveVpnPermission -> onOnboardingDone()
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/onboarding/ui/page/vpn/VpnPermissionPage.kt:129]: As this has been removed we can clean up OnboardingPageFragment. The `fun onOnboardingDone()` declared there is not needed anymore.

File: app/src/main/java/com/duckduckgo/app/onboarding/ui/page/vpn/VpnPagesViewModel.kt [hunk 2]
```diff
@@ -1,215 +0,0 @@
...
-    private fun sendCommand(newCommand: Command) {
-        viewModelScope.launch {
-            command.send(newCommand)
-        }
-    }
-
-    private fun introPageBecameVisible() {
-        if (!viewHasShown) {
-            viewHasShown = true
-            pixel.fire(AppPixelName.ONBOARDING_VPN_INTRO_SHOWN)
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/onboarding/ui/page/vpn/VpnPagesViewModel.kt:92]: Should we clean up `ONBOARDING_VPN_INTRO_SHOWN` from AppPixelName?

Other pixels in the same category:
`ONBOARDING_VPN_PERMISSION_SHOWN`
`ONBOARDING_VPN_INTRO_CONTINUED`
`ONBOARDING_VPN_INTRO_SKIPPED`
`ONBOARDING_VPN_PERMISSION_SKIPPED`
`ONBOARDING_VPN_PERMISSION_FAQ_LAUNCHED`
`ONBOARDING_VPN_PERMISSION_CONTINUED`
`ONBOARDING_VPN_PERMISSION_LAUNCHED`
`ONBOARDING_VPN_PERMISSION_DENIED`

File: app/src/test/java/com/duckduckgo/app/survey/api/SurveyDownloaderTest.kt [hunk 3]
```diff
@@ -117,37 +115,6 @@ class SurveyDownloaderTest {
         verify(mockDao).deleteUnusedSurveys()
     }
 
-    @Test
-    fun whenAppTpSurveyContainsRetentionStudyControlSurveyAndUserInRetentionStudyThenSurveyScheduled() {
-        val mockAppTPCall = mock<Call<SurveyGroup?>>()
-        whenever(mockAppTPCall.execute()).thenReturn(Response.success(surveyWithAllocationForAppTPRetentionStudy("abc")))
```
Comment (anikiki) [inline app/src/test/java/com/duckduckgo/app/survey/api/SurveyDownloaderTest.kt:123]: `surveyWithAllocationForAppTPRetentionStudy` function can be removed.

We may need to clean up SurveyGroup.SurveyOption as I noticed inside `surveyWithAllocationForAppTPRetentionStudy` we create a SurveyOption with `isAtpRetentionStudyRequired` which is not needed anymore.

**Review by anikiki — APPROVED** [review]
Looks good! Left a few comments about unused code. Tested and didn't find anything unusual.

File: app/src/main/java/com/duckduckgo/app/onboarding/ui/page/vpn/VpnPermissionPage.kt (hunk 1 above)
Comment (aitorvs) [reply app/src/main/java/com/duckduckgo/app/onboarding/ui/page/vpn/VpnPermissionPage.kt:129]: Will do

File: app/src/main/java/com/duckduckgo/app/onboarding/ui/page/vpn/VpnPagesViewModel.kt (hunk 2 above)
Comment (aitorvs) [reply app/src/main/java/com/duckduckgo/app/onboarding/ui/page/vpn/VpnPagesViewModel.kt:92]: Will do!

File: app/src/test/java/com/duckduckgo/app/survey/api/SurveyDownloaderTest.kt (hunk 3 above)
Comment (aitorvs) [reply app/src/test/java/com/duckduckgo/app/survey/api/SurveyDownloaderTest.kt:123]: Yup, will do!

**Pull request merged**